components/ws-manager-bridge/src/cluster-service-server.ts

```diff
@@ -70,21 +70,22 @@ export class ClusterService implements IClusterServiceServer {
             try {
                 // check if the name or URL are already registered/in use
                 const req = call.request.toObject();
-                await Promise.all([
-                    async () => {
-                        const oldCluster = await this.clusterDB.findByName(req.name);
-                        if (!oldCluster) {
-                            throw new GRPCError(grpc.status.ALREADY_EXISTS, `a WorkspaceCluster with name ${req.name} already exists in the DB`);
-                        }
-                    },
-                    async () => {
-                        const oldCluster = await this.clusterDB.findFiltered({ url: req.url });
-                        if (!oldCluster) {
-                            throw new GRPCError(grpc.status.ALREADY_EXISTS, `a WorkspaceCluster with url ${req.url} already exists in the DB`);
-                        }
-                    }
+
+                const clusterByNamePromise = this.clusterDB.findByName(req.name);
+                const clusterByUrlPromise = this.clusterDB.findFiltered({ url: req.url })
+
+                const [clusterByName, clusterByUrl] = await Promise.all([
+                    clusterByNamePromise,
+                    clusterByUrlPromise
                 ]);
 
+                if (!!clusterByName) {
+                    throw new GRPCError(grpc.status.ALREADY_EXISTS, `a WorkspaceCluster with name ${req.name} already exists in the DB`);
+                }
+                if (Array.isArray(clusterByUrl) && clusterByUrl.length > 0) {
```

Comment:
hm. what if `findFiltered` will change in the future and will return something other than array?

Reply:
Good point, since it's TypeScript this is overkill, null check and length check should be good. Thanks!